### PR TITLE
[#2678] improvement(test): Add e2e test for jdbc backend

### PR DESCRIPTION
**Pull request by yuqi1129**
### What changes were proposed in this pull request?

Add JDBC entity store in integration test. 

### Why are the changes needed?

To test whether JDBC backends works.

Fix: #2678 

### Does this PR introduce _any_ user-facing change?

N/A.

### How was this patch tested?

Existing ITs

### CODE REVIEW

File: core/src/main/java/com/datastrato/gravitino/catalog/CatalogOperationDispatcher.java [hunk 1]
```diff
@@ -546,6 +551,8 @@ public Table alterTable(NameIdentifier ident, TableChange... changes)
                         t.loadTable(NameIdentifier.of(ident.namespace(), tempAlteredTable.name()))),
             NoSuchTableException.class);
 
+    LOG.info("yuqiabc: alterTable: ident: {}, alteredTable: {}", ident, alteredTable.auditInfo());
```
Comment (FANNG1) [inline core/src/main/java/com/datastrato/gravitino/catalog/CatalogOperationDispatcher.java:554]: typo..

Comment (yuqi1129) [reply core/src/main/java/com/datastrato/gravitino/catalog/CatalogOperationDispatcher.java:554]: Sorry, this PR is still in draft form...

File: core/src/main/java/com/datastrato/gravitino/storage/relational/mapper/SchemaMetaMapper.java [hunk 2]
```diff
@@ -120,7 +120,7 @@ SchemaPO selectSchemaMetaByCatalogIdAndName(
           + " AND schema_name = #{oldSchemaMeta.schemaName}"
           + " AND metalake_id = #{oldSchemaMeta.metalakeId}"
           + " AND catalog_id = #{oldSchemaMeta.catalogId}"
-          + " AND schema_comment = #{oldSchemaMeta.schemaComment}"
+          + " AND (schema_comment IS NULL OR schema_comment = #{oldSchemaMeta.schemaComment})"
```
Comment (yuqi1129) [inline core/src/main/java/com/datastrato/gravitino/storage/relational/mapper/SchemaMetaMapper.java:123]: This is to fix the problem that `schema_comment` is nullable

**Comment by yuqi1129** [comment]
After this PR, the backend integration test will take as long as 50 minutes.

**Comment by yuqi1129** [comment]
@xloya 
Can you also assist in taking a look?

File: integration-test-common/src/test/java/com/datastrato/gravitino/integration/test/util/AbstractIT.java [hunk 3]
```diff
@@ -98,6 +110,48 @@ private static void recoverGravitinoServerConfig() throws IOException {
...
+    if (!ITUtils.EMBEDDED_TEST_MODE.equals(testMode)) {
+      String gravitinoHome = System.getenv("GRAVITINO_HOME");
+      java.nio.file.Path tmpPath = Paths.get(gravitinoHome, relativeDeployLibsPath);
+      JdbcDriverDownloader.downloadJdbcDriver(DOWNLOAD_JDBC_DRIVER_URL, tmpPath.toString());
+    }
+  }
+
+  private static void setMySQLBackend() {
+    String mysqlUrl = MYSQL_CONTAINER.getJdbcUrl();
+    customConfigs.put(Configs.ENTITY_STORE.getKey(), "relational");
```
Comment (xloya) [inline integration-test-common/src/test/java/com/datastrato/gravitino/integration/test/util/AbstractIT.java:123]: How about use Configs.RELATIONAL_ENTITY_STORE?

File: integration-test-common/src/test/java/com/datastrato/gravitino/integration/test/util/AbstractIT.java [hunk 4]
```diff
@@ -98,6 +110,48 @@ private static void recoverGravitinoServerConfig() throws IOException {
...
+    customConfigs.put(Configs.ENTITY_RELATIONAL_STORE.getKey(), "JDBCBackend");
```
Comment (xloya) [inline integration-test-common/src/test/java/com/datastrato/gravitino/integration/test/util/AbstractIT.java:124]: Configs.DEFAULT_ENTITY_RELATIONAL_STORE.

File: integration-test-common/src/test/java/com/datastrato/gravitino/integration/test/util/AbstractIT.java [hunk 5]
```diff
@@ -98,6 +110,48 @@ private static void recoverGravitinoServerConfig() throws IOException {
...
+    customConfigs.put(Configs.ENTITY_RELATIONAL_JDBC_BACKEND_USER.getKey(), "root");
+    customConfigs.put(Configs.ENTITY_RELATIONAL_JDBC_BACKEND_PASSWORD.getKey(), "root");
+
+    LOG.info("MySQL URL: {}", mysqlUrl);
+    // Connect to the mysql docker and create a databases
+    try (Connection connection =
+        DriverManager.getConnection(
+            StringUtils.substring(mysqlUrl, 0, mysqlUrl.lastIndexOf("/")), "root", "root")) {
+
+      final Statement statement = connection.createStatement();
```
Comment (xloya) [inline integration-test-common/src/test/java/com/datastrato/gravitino/integration/test/util/AbstractIT.java:137]: Maybe we can put this line in the try-with-resources statement to close statement automatically.

Comment (yuqi1129) [reply integration-test-common/src/test/java/com/datastrato/gravitino/integration/test/util/AbstractIT.java:137]: OK

File: core/src/main/java/com/datastrato/gravitino/storage/relational/mapper/SchemaMetaMapper.java (hunk 2 above)
Comment (xloya) [reply core/src/main/java/com/datastrato/gravitino/storage/relational/mapper/SchemaMetaMapper.java:123]: I'm wondering if it is necessary for us to convert the `null` value to `""` while inserting, to solve this problem

Comment (xloya) [reply core/src/main/java/com/datastrato/gravitino/storage/relational/mapper/SchemaMetaMapper.java:123]: Should we set this parameter to `Not Null` and use `""` as the default value?

**Comment by xloya** [comment]
> After this PR, the backend integration test will take as long as 50 minutes.

I'm not sure if it's because of the driver download. During my internal testing, downloading the mysql driver from `https://repo1.maven.org/maven2` was very slow.

File: core/src/main/java/com/datastrato/gravitino/storage/relational/service/TableMetaService.java [hunk 6]
```diff
@@ -21,7 +21,6 @@
 import java.util.Objects;
 import java.util.function.Function;
 
-/** The service class for table metadata. It provides the basic database operations for table. */
```
Comment (xloya) [inline core/src/main/java/com/datastrato/gravitino/storage/relational/service/TableMetaService.java:24]: Is this necessary?

**Comment by yuqi1129** [comment]
> > After this PR, the backend integration test will take as long as 50 minutes.
> 
> I'm not sure if it's because of the driver download. During my internal testing, downloading the mysql driver from `https://repo1.maven.org/maven2` was very slow.

I'm afraid 50 minutes is as expected. Previous ITs took about 23 minutes on average, as this PR adds another test mode, the total time I think should be double, moreover, the MySQL entity store should be more time-consuming that the KV entity store. 

Downdowing jars in CI are very fast as they won't be blocked by GFW.

File: .github/workflows/backend-integration-test.yml [hunk 7]
```diff
@@ -83,6 +83,7 @@ jobs:
       - name: Backend Integration Test
         id: integrationTest
         run: |
+          ./gradlew test --rerun-tasks -PskipTests -PtestMode=${{ matrix.test-mode }} -PjdkVersion=${{ matrix.java-version }} -PskipWebITs -PjdbcBackend
```
Comment (xloya) [inline .github/workflows/backend-integration-test.yml:86]: Is it possible for us to implement parallel IT here?

File: core/src/main/java/com/datastrato/gravitino/storage/relational/service/TableMetaService.java (hunk 6 above)
Comment (yuqi1129) [reply core/src/main/java/com/datastrato/gravitino/storage/relational/service/TableMetaService.java:24]: I also noticed this and Iplan to remove it.

File: core/src/main/java/com/datastrato/gravitino/storage/relational/mapper/SchemaMetaMapper.java (hunk 2 above)
Comment (yuqi1129) [reply core/src/main/java/com/datastrato/gravitino/storage/relational/mapper/SchemaMetaMapper.java:123]: Comment null and empty are different for many catalogs, so I don't think it's better than the previous design.

**Comment by yuqi1129** [comment]
@jerryshao Please help to review it if you have time, thanks

**Comment by yuqi1129** [comment]
> @jerryshao Please help to review it if you have time, thanks

Could you please help to review it, thanks

**Comment by xloya** [comment]
Overall LGTM.

**Comment by jerryshao** [comment]
I have no further comment, my simple worry is that it will add more CI jobs that make the CI time much longer.

**Review by jerryshao — APPROVED** [review]

**Pull request merged**